tests/lib/mem_blocks/src/main.c

```diff
@@ -205,6 +205,182 @@ static void test_mem_block_multi_alloc_free_alt_buf(void)
 	alloc_free(&mem_block_02, NUM_BLOCKS, 10);
 }
 
+static void test_mem_block_get(void)
+{
+	int i, ret, val;
+	void *block, *block1;
+
+#ifdef CONFIG_SYS_MEM_BLOCKS_LISTENER
+	listener_idx = 0;
+	heap_listener_register(&mem_block_01_alloc);
+	heap_listener_register(&mem_block_01_free);
+#endif
+
+	/* get a 2 entiries memory block starting from 0 */
+	ret = sys_mem_blocks_get(&mem_block_01, 0, 2, &block);
+	zassert_equal(ret, 0,
+		      "sys_mem_blocks_get failed (%d)", ret);
+
+	/* blocks 0 and 1 should be taken */
+	for (i = 0; i < NUM_BLOCKS; i++) {
+		ret = sys_bitarray_test_bit(mem_block_01.bitmap,
+					    i, &val);
+		switch (i) {
+		case 0:
+		case 1:
```

Comment:
Style: seems like a needlessly complicated switch vs. something like:

```c
if (i < 2) {
	zassert_equal(val, 1,
		      "sys_mem_blocks_get bitmap failed, bit %i should be set", i);
} else {
	zassert_equal(val, 0,
		      "sys_mem_blocks_get bitmap failed, bit %i should be cleared", i);
}
```

Reply:
yes and no. I intentionally made it with case, to explicitly show which blocks should be taken/free. It's getting more complicated in other tests (next commit)  - and ifs like in your comment would become hard to read. All of tests and verifications are prepared in the way it would run properly regardless of NUM_BLOCKS value